### PR TITLE
[WFLY-15288] Switch to using org.jboss.modules.ClassTransformer inste…

### PR DESCRIPTION
**Pull request by xjusko**
…ad of j.l.i.ClassFileTransformer.

fixes https://issues.redhat.com/browse/WFLY-15288

Issue is related to https://issues.redhat.com/browse/WFCORE-3685 so it wont be compiled.

### CODE REVIEW

File: jpa/hibernate-transformer/src/main/java/org/jboss/as/hibernate/Hibernate51CompatibilityTransformer.java [hunk 1]
```diff
@@ -43,13 +43,14 @@
 import org.objectweb.asm.Opcodes;
 import org.objectweb.asm.util.TraceClassVisitor;
 import org.wildfly.security.manager.WildFlySecurityManager;
+import org.jboss.modules.ClassTransformer;
```
Comment (bstansberry) [inline jpa/hibernate-transformer/src/main/java/org/jboss/as/hibernate/Hibernate51CompatibilityTransformer.java:46]: Nit: Please put this earlier in the block, in alphabetical order.

**Comment by bstansberry** [comment]
Thanks, @xjusko, but @scottmarlow 's plan is to remove these classes. See #14900

**Comment by scottmarlow** [comment]
> Thanks, @xjusko, but @scottmarlow 's plan is to remove these classes. See #14900

@bstansberry this pr should definitely be updated to not update the Hibernate ORM 5.1 => 5.3 bytecode transformer but we still need some of the changes is in this pull request, specifically the following files need updating to prepare for https://github.com/wildfly/wildfly-core/pull/4733 (`Use org.jboss.modules.ClassTransformer for transformation DUPs`):
- jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPAClassFileTransformerProcessor.java
- jpa/subsystem/src/main/java/org/jboss/as/jpa/processor/JPADelegatingClassFileTransformer.java
- testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/classloading/transformer/DummyClassFileTransformer1.java
- testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/deployment/classloading/transformer/DummyClassFileTransformer2.java

**Comment by bstansberry** [comment]
Ok, I'll re-open this.

**Comment by jmesnil** [comment]
@xjusko this PR is superseded by #15484 where I cherry-picked your commit. Thanks!

**Pull request closed without merging**